elastic/kibana
# [Logs UI] Use create over update for log source configuration updates

## Summary

This is a small fix / followup to #96454. It changes `update` to `create` with `overwrite: true` due to the deep merge nature of `update`, which leaves unnecessary properties hanging around.

## Testing

- Delete your `infrastructure-ui-source:default` saved object if needed (`app/management/kibana/objects`)
- Start out by saving a change from the settings page, using the index name format. Don't change to a KIP yet. This creates our saved object.
- Now update to using a KIP via the settings page.
- Check the properties on the saved object:

The following in dev tools will work (there's also a Saved Objects API):

```
GET .kibana_8.0.0/_search
{
  "query": {
    "terms": {
      "_id": [
        "infrastructure-ui-source:default"
      ]
    }
  }
}
```

You should **not** see a merged set of properties like the following on the `logIndices` property:

```
"logIndices": {
  "type": "index_pattern",
  "indexPatternId": "ca1dd290-97bc-11eb-8ea8-c11052d2b652",
  "indexName": "filebeat-*,logs-*" // < bad
}
```

Instead the `logIndices` property should only represent the current type.
